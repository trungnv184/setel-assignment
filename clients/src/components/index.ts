export { default as Header } from './Header/Header';
export { default as ProductItem } from './ProductItem/ProductItem';
export { default as ProductList } from './ProductList/ProductList';
export { default as Layout } from './Layout/Layout';
export { default as CartList } from './Cart/CartList';
export { default as MenuBar } from './MenuBar/MenuBar';
export { default as ErrorMessage } from './ErrorMessage/ErrorMessage';
export { default as OrderForm } from './OrderForm/OrderForm';
export { StyledAppContainer } from './Common/Style';
