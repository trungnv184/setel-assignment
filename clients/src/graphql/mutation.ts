import { gql } from '@apollo/client';

const ADD_PRODUCT_TO_CART = gql`
  mutation AddToCart($cartInput: CartInput) {
    addToCart(cartInput: $cartInput) {
      id
      name
      url
      price
    }
  }
`;

const CREATE_ORDER_PRODUCTS = gql`
  mutation CreateOrder($orderInput: OrderInput) {
    createOrder(orderInput: $orderInput) {
      orderItems {
        id
        name
        url
        price
        quantity
        createdDate
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
    }
  }
`;

export { ADD_PRODUCT_TO_CART, CREATE_ORDER_PRODUCTS };
