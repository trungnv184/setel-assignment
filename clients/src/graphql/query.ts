import { gql } from '@apollo/client';

const GET_PRODUCTS = gql`
  query GetProducts {
    products {
      id
      name
      price
      url
      rating
    }
  }
`;

const GET_PRODUCTS_IN_CART = gql`
  query GetProductsCart {
    getProductsCart {
      id
      name
      price
      url
      quantity
    }
  }
`;

const GET_ORDER_BY_ID = gql`
  query GetOrder($orderId: String) {
    getOrder(orderId: $orderId) {
      id
      orderItems {
        id
        name
        price
        url
        quantity
      }
      metadata {
        firstName
        lastName
        address
        phoneNumber
      }
      notes
      state
      createdDate
      updatedDate
    }
  }
`;

const GET_ORDER_STATUS = gql`
  query GetOrderStatus($orderId: String) {
    getOrder(orderId: $orderId) {
      state
    }
  }
`;

export { GET_PRODUCTS, GET_PRODUCTS_IN_CART, GET_ORDER_BY_ID, GET_ORDER_STATUS };
