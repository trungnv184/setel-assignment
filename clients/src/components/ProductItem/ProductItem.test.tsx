import { render, screen, waitFor } from '@testing-library/react';
import { ProductItem } from 'components';
import { MockedProvider } from '@apollo/client/testing';
import { PRODUCTS_MOCK } from 'test/mocks/products.mock';
import { ADD_PRODUCT_TO_CART } from 'graphql/mutation';

const mocks = [
  {
    request: {
      query: ADD_PRODUCT_TO_CART,
      variables: {
        cartInput: {
          name: 'Mock Name',
          url: 'Mock url',
          price: 500,
        },
      },
    },
    result: () => {
      return {
        data: {
          addToCart: {
            id: 10000,
            name: 'Architecture Design',
            url: 'link-to-book',
            price: 5000,
            quantity: 1,
            createdDate: '2021-07-02',
          },
        },
      };
    },
  },
];
describe('ProductItem', () => {
  test('should match snapshot', async () => {
    const { name, url, price, rating } = PRODUCTS_MOCK[0];
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ProductItem name={name} url={url} price={price} rating={rating} />
      </MockedProvider>
    );

    const button = screen.getByText(/Buy/i);

    button.click();

    await waitFor(() => {
      const tooltipMessage = screen.getByText(/Your product have been added to cart!/i);
      expect(tooltipMessage).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
