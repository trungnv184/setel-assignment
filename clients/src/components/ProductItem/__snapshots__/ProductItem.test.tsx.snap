// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem should match snapshot 1`] = `
<div>
  <div
    class="ui raised card"
  >
    <div
      class="large image"
    >
      <img
        src="https://salt.tikicdn.com/cache/280x280/ts/product/6e/41/a2/721878ef916620dbcc7ff569adcd577d.jpg"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="header"
        data-testid="book-name"
      >
        Kiên trì hay từ bỏ
      </div>
      <div
        class="meta"
      >
        <strong>
          200.000 ₫
        </strong>
      </div>
    </div>
    <div
      class="extra content"
    >
      <div
        class="ui star rating"
        role="radiogroup"
        tabindex="-1"
      >
        <i
          aria-checked="false"
          aria-posinset="1"
          aria-setsize="5"
          class="active icon"
          role="radio"
          tabindex="0"
        />
        <i
          aria-checked="false"
          aria-posinset="2"
          aria-setsize="5"
          class="active icon"
          role="radio"
          tabindex="0"
        />
        <i
          aria-checked="false"
          aria-posinset="3"
          aria-setsize="5"
          class="active icon"
          role="radio"
          tabindex="0"
        />
        <i
          aria-checked="false"
          aria-posinset="4"
          aria-setsize="5"
          class="active icon"
          role="radio"
          tabindex="0"
        />
        <i
          aria-checked="true"
          aria-posinset="5"
          aria-setsize="5"
          class="active icon"
          role="radio"
          tabindex="0"
        />
      </div>
      <button
        class="ui red right floated button"
      >
        Buy
      </button>
    </div>
  </div>
</div>
`;
